streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -178,6 +180,26 @@ public void testProcessOrder() throws Exception {
 
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldAddTimestampExtractorPerSource() throws Exception {
```

Comment:
I agree with @dguy If you test `TopologyBuilder#addSource()` it should go to `TopologyBuilderTest`, and if you test `KStreamBuilder#stream` it should go to `KStreamBuilderTest` -- this also implies, you should split this test into two.

Also testing `KStreamBuilder#addSource` is redundant because its inherited from `TopologyBuilder`.